mm2src/coins/lp_coins.rs

```diff
@@ -5542,18 +5542,42 @@ pub async fn set_requires_notarization(ctx: MmArc, req: Json) -> Result<Response
 
 pub async fn show_priv_key(ctx: MmArc, req: Json) -> Result<Response<Vec<u8>>, String> {
     let ticker = try_s!(req["coin"].as_str().ok_or("No 'coin' field")).to_owned();
-    let coin = match lp_coinfind(&ctx, &ticker).await {
-        Ok(Some(t)) => t,
-        Ok(None) => return ERR!("No such coin: {}", ticker),
-        Err(err) => return ERR!("!lp_coinfind({}): {}", ticker, err),
-    };
-    let res = try_s!(json::to_vec(&json!({
-        "result": {
-            "coin": ticker,
-            "priv_key": try_s!(coin.display_priv_key()),
-        }
-    })));
-    Ok(try_s!(Response::builder().body(res)))
+
+    // Try to get activated coin first (existing behavior for backwards compatibility)
+    match lp_coinfind(&ctx, &ticker).await {
+        Ok(Some(coin)) => {
+            let res = try_s!(json::to_vec(&json!({
+                "result": {
+                    "coin": ticker,
+                    "priv_key": try_s!(coin.display_priv_key()),
+                }
+            })));
+            Ok(try_s!(Response::builder().body(res)))
+        },
+        Ok(None) | Err(_) => {
+            use crate::rpc_command::offline_keys::{offline_keys_export_internal, OfflineKeysRequest};
+
+            let offline_req = OfflineKeysRequest {
+                coins: vec![ticker.clone()],
+            };
+            match offline_keys_export_internal(ctx, offline_req).await {
+                Ok(response) => {
+                    if let Some(coin_info) = response.result.first() {
+                        let res = try_s!(json::to_vec(&json!({
+                            "result": {
+                                "coin": ticker,
+                                "priv_key": coin_info.priv_key,
+                            }
+                        })));
+                        Ok(try_s!(Response::builder().body(res)))
+                    } else {
+                        ERR!("No key info found for coin: {}", ticker)
+                    }
+                },
+                Err(err) => ERR!("Failed to derive offline key for {}: {}", ticker, err),
+            }
+        },
+    }
```

Comment:
This fallback logic may cause confusion in HD mode, as If a coin is inactive, this new fallback logic will execute `offline_keys_export_internal`, which derives the legacy Iguana private key directly from the seed that is used for an HD wallet. We should keep the old `show_priv_key` method as it was before this PR, remove `KeyExportMode::standard` from the new `get_private_keys` RPC Request, then remove `offline_keys_export_internal` completely. There should be no Default `KeyExportMode` after these changes as I will instruct on how to deal with that in another review comment.